src/main/java/spoon/reflect/factory/TypeFactory.java

```diff
@@ -324,6 +324,9 @@ public TypeFactory() {
 		}
 
 		ref.setSimpleName(type.getSimpleName());
+		for (CtTypeParameter typeParam : type.getFormalCtTypeParameters()) {
```

Comment:
I have already found, that this change is wrong. I will rollback it and do it different.